libraries/Microsoft.Bot.Builder.Dialogs/DialogContainer.cs

```diff
@@ -53,7 +54,24 @@ protected virtual string GetInternalVersion()
         /// 
         /// This should be called at the start of `beginDialog()`, `continueDialog()`, and `resumeDialog()`.
         /// </remarks>
-        protected virtual async Task CheckForVersionChangeAsync(DialogContext dc)
+        protected virtual Task CheckForVersionChangeAsync(DialogContext dc)
```

Comment:
this is a bit weird - why not plumb through the cancellation

Reply:
I agree

---
In reply to: [422350766](https://github.com/microsoft/botbuilder-dotnet/pull/3892#discussion_r422350766) [](ancestors = 422350766)

Reply:
Oh because it is in the dialogs assembly and it is abreaking change to binary level backward compat

Reply:
no this is new method, we should fix.


---
In reply to: [422385918](https://github.com/microsoft/botbuilder-dotnet/pull/3892#discussion_r422385918) [](ancestors = 422385918)